### PR TITLE
Added merkle tree schema

### PR DESCRIPTION
**Pull request by rg911**
Added merkle tree schema

### CODE REVIEW

File: spec/schemas/MerkleTreeNodeTypeEnum.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+type: integer
+enum:
+  - 0
+  - 255
+description: |
+  Type of merkle tree node type:
```
Comment (segfaultxavi) [inline spec/schemas/MerkleTreeNodeTypeEnum.yml:6]:
```suggestion
  Type of merkle tree node:
```

File: spec/schemas/MerkleTreeLeaf.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - type
+  - path
+  - encodedPath
+  - nibbleCount
+  - value
+  - leafHash
+properties:
+  type:
+    description: Merkle tree node type.
+    $ref: "./MerkleTreeNodeTypeEnum.yml"
```
Comment (segfaultxavi) [inline spec/schemas/MerkleTreeLeaf.yml:13]: It's nice to have these `description`s but I think they will be ignored due to #233.

**Review by segfaultxavi — DISMISSED** [review]
Isn't **Merkle** a proper noun? Shouldn't it be capitalized everywhere?

File: spec/schemas/MerkleStateInfoDTO.yml [hunk 3]
```diff
@@ -3,6 +3,14 @@ description: |
   The merkle path information clients can use to proof the state of the given entity.
 required:
   - raw
+  - tree
 properties:
   raw:
     $ref: "./MerkleTreeRaw.yml"
+  tree:
```
Comment (fboucquez) [inline spec/schemas/MerkleStateInfoDTO.yml:10]: I would split tree into branches (list of MerkleTreeBranchDTO) and leaf (optional MerkleTreeLeafDTO) removing the anyOf and simplifying the mappers into 1-1 in the SDKs

**Review by fboucquez — CHANGES_REQUESTED** [review]

File: spec/schemas/MerkleStateInfoDTO.yml (hunk 3 above)
Comment (fboucquez) [reply spec/schemas/MerkleStateInfoDTO.yml:10]: Only if it makes sense, we only have 1 leaf at most.

Comment (fboucquez) [reply spec/schemas/MerkleStateInfoDTO.yml:10]: After last conversation, we have decided to go with a generic list of branches/leaves. Only one leaf is expected in a valid case, but it's good to have a generic structure

**Review by fboucquez — APPROVED** [review]

File: spec/schemas/MerkleTreeLeaf.yml (hunk 2 above)
Comment (fboucquez) [reply spec/schemas/MerkleTreeLeaf.yml]: We will need to revisit the descriptions next to $ref

**Review by fboucquez — APPROVED** [review]

**Pull request merged**